File: BE.Core.FW/Backend/Business/Navigation/NavigationModel.cs
namespace Backend.Business.Navigation
{
    public class NavigationModel
    {
        public Guid Id { get; set; }
        public Guid? ParentId { get; set; }
        public string Code { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public string IconClass { get; set; } = string.Empty;
        public int Order { get; set; }
        public string Resource { get; set; } = string.Empty;
        public string ComponentPath { get; set; } = string.Empty;
        public List<NavigationModel> Children { get; set; } = new List<NavigationModel>();
        public bool IsShow { get; set; }
    }
}

File: FE.Core.FW/src/apis/models/NavigationModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type NavigationModel = {
  id?: string;
  parentId?: string | null;
  code?: string | null;
  name?: string | null;
  url?: string | null;
  iconClass?: string | null;
  order?: number;
  resource?: string | null;
  componentPath?: string | null;
  children?: Array<NavigationModel> | null;
  isShow?: boolean
};

export type NavigationModelCustom = {
  key?: string;
  parentId?: string | null;
  code?: string | null;
  name?: string | null;
  url?: string | null;
  isShow?: boolean;
  iconClass?: string | null;
  order?: number;
  resource?: string | null;
  componentPath?: string | null;
  children?: NavigationModelCustom[] | null;
};
